test/integration-tests/RemoteAdapterAdd.t.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity >=0.8.9;
+
+/// library imports
+import {Vm} from "forge-std/Test.sol";
+
+/// local imports
+import "test/Setup.t.sol";
+
+import {MultiMessageSender} from "src/MultiMessageSender.sol";
+import {MultiMessageReceiver} from "src/MultiMessageReceiver.sol";
+import {Error} from "src/libraries/Error.sol";
+import {GovernanceTimelock} from "src/controllers/GovernanceTimelock.sol";
+
+/// @dev scenario: admin updates sender adapters on dst chain using message from source chain
+/// @notice handles both single add and multiple add
+contract RemoteAdapterAdd is Setup {
+    /// @dev intializes the setup
+    function setUp() public override {
+        super.setUp();
+    }
+
+    /// @dev just add one adapter and assert
+    function test_remoteAddReceiverAdapterSingle() public {
+        address[] memory adaptersToAdd = new address[](1);
+        adaptersToAdd[0] = address(420421422);
+
+        /// true = add
+        /// false = remove
+        bool[] memory operation = new bool[](1);
+        operation[0] = true;
+
+        _adapterAdd(adaptersToAdd, operation);
+    }
+
+    /// @dev add multiple adapters and assert
+    function test_remoteAddReceiverAdapterMulti() public {
+        address[] memory adaptersToAdd = new address[](3);
+        adaptersToAdd[0] = address(42042142232313);
+        adaptersToAdd[1] = address(22132131);
+        adaptersToAdd[2] = address(22132132131);
+
+        /// true = add
+        /// false = remove
+        bool[] memory operation = new bool[](3);
+        operation[0] = true;
+        operation[1] = true;
+        operation[2] = true;
+
+        _adapterAdd(adaptersToAdd, operation);
+    }
+
+    function _adapterAdd(address[] memory adaptersToAdd, bool[] memory operation) internal {
+        vm.selectFork(fork[1]);
+        vm.startPrank(caller);
+
+        /// send cross-chain message using MMA infra
+        vm.recordLogs();
+        MultiMessageSender(contractAddress[1][bytes("MMA_SENDER")]).remoteCall{value: 2 ether}(
+            137,
+            address(contractAddress[137][bytes("MMA_RECEIVER")]),
+            abi.encodeWithSelector(MultiMessageReceiver.updateReceiverAdapter.selector, adaptersToAdd, operation),
```

Comment:
nit: rename `MultiMessageReceiver.updateReceiverAdapter(...)` to its plural form `MultiMessageReceiver.updateReceiverAdapters(...)`